.github/workflows/docker-push.yml

```diff
@@ -0,0 +1,83 @@
+---
+name: 'Push Docker Image'
+on:
+  # TODO: remove, only for testing; replace with
+  # workflow_dispatch:
+  push:
+    branches:
+      - '**' # Trigger on every branch
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  dockerhub:
+    name: 'Build and Publish Docker Image'
+    runs-on: [self-hosted, linux, normal]
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.push.head.sha }}
+          fetch-depth: 0
+
+      - name: 'Set environment'
+        run: |
+          KONTROL_VERSION=$(cat package/version)
+          echo "CONTAINER_NAME=kontrol-ci-docker-${GITHUB_SHA}" >> ${GITHUB_ENV}
+          BRANCH_NAME="${{ github.ref_name }}"
```

Comment:
```
BRANCH_NAME="${{ github.events.inputs.kontrol_branch }}"
```

Reply:
Thanks! Done in https://github.com/runtimeverification/kontrol/pull/913/commits/aa2db2252810d2322637286a2211187699909bb4.